core/src/main/java/org/apache/calcite/sql/dialect/DuckDBSqlDialect.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.dialect;
+
+import org.apache.calcite.avatica.util.TimeUnitRange;
+import org.apache.calcite.config.NullCollation;
+import org.apache.calcite.rel.type.RelDataTypeSystem;
+import org.apache.calcite.rel.type.RelDataTypeSystemImpl;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlDialect;
+import org.apache.calcite.sql.SqlLiteral;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.fun.SqlLibraryOperators;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+/**
+ * A <code>SqlDialect</code> implementation for the DuckDB database.
+ */
+public class DuckDBSqlDialect extends StarRocksSqlDialect {
+  public static final RelDataTypeSystem TYPE_SYSTEM =
+      new RelDataTypeSystemImpl() {
+
+        // We can refer to document of DuckDB 1.2.x:
+        // https://duckdb.org/docs/stable/sql/data_types/numeric#fixed-point-decimals
+        @Override public int getMaxPrecision(SqlTypeName typeName) {
+          switch (typeName) {
+          case DECIMAL:
+            return 38;
+          default:
+            return super.getMaxPrecision(typeName);
+          }
+        }
+
+        @Override public int getMaxScale(SqlTypeName typeName) {
+          switch (typeName) {
+          case DECIMAL:
+            return 38;
+          default:
+            return super.getMaxScale(typeName);
+          }
+        }
+
+        @Override public int getMaxNumericScale() {
+          return getMaxScale(SqlTypeName.DECIMAL);
+        }
+      };
+
+  public static final SqlDialect.Context DEFAULT_CONTEXT = SqlDialect.EMPTY_CONTEXT
+      .withDatabaseProduct(SqlDialect.DatabaseProduct.DUCKDB)
+      .withIdentifierQuoteString("\"")
+      .withNullCollation(NullCollation.LAST)
```

Comment:
Is this LAST or HIGH?
These are not the same.

Reply:
From definition：
LAST：Nulls last for DESC and ASC.
HIGH：Nulls first for DESC, nulls last for ASC.

system env test：
before query
insert into t3 select 1,'aa';
insert into t3 select 1,'bb';
insert into t3 select 1,NULL;
insert into t3 select 1,NULL;

then query order by desc/asc
D select * from t3 order by data asc;
┌───────┬─────────┐
│  id   │  data   │
│ int32 │ varchar │
├───────┼─────────┤
│     1 │ aa      │
│     1 │ bb      │
│     1 │         │
└───────┴─────────┘
D select * from t3 order by data desc;
┌───────┬─────────┐
│  id   │  data   │
│ int32 │ varchar │
├───────┼─────────┤
│     1 │ bb      │
│     1 │ aa      │
│     1 │         │
└───────┴─────────┘

we can find that NULL is always at last, so here should be LAST.

Reply:
Can you add a link to https://duckdb.org/docs/stable/sql/query_syntax/orderby.html so that people can check easily?

Reply:
OK, had added the link to the code~ @mihaibudiu